analyzers/tests/SonarAnalyzer.UnitTest/TestCases/SymbolicExecution/Roslyn/EmptyCollectionsShouldNotBeEnumerated.cs

```diff
@@ -3,617 +3,391 @@
 using System.Collections.ObjectModel;
 using System.Collections.Specialized;
 using System.Linq;
+using System.Runtime.Serialization;
 using System.Text;
 
-class ListTests
+class CollectionTests
 {
     private IEnumerable<int> items = new List<int> { 1, 2, 3 };
+    private IDictionary<int, int> dictionaryItems = new Dictionary<int, int>();
 
     private static bool Predicate(int i) => true;
     private static void Action(int i) { }
-    private List<int> Factory() => new List<int>();
+    private List<int> ListFactory() => new List<int>();
+    private HashSet<int> SetFactory() => new HashSet<int>();
+    private Queue<int> QueueFactory() => new Queue<int>();
+    private Stack<int> StackFactory() => new Stack<int>();
+    private ObservableCollection<int> ObservableCollectionFactory() => new ObservableCollection<int>();
+    private int[] ArrayFactory() => new int[0];
+    private Dictionary<int, int> DictionaryFactory() => new Dictionary<int, int>();
 
-    public void DefaultConstructor_Applies_Empty()
+    public void DefaultConstructor()
     {
         var list = new List<int>();
-        list.Clear(); // FIXME Non-compliant {{Remove this call, the collection is known to be empty here.}}
-//      ~~~~~~~~~~~~
+        list.Clear();   // Noncompliant {{Remove this call, the collection is known to be empty here.}}
+//      ^^^^^^^^^^^^
+        var set = new HashSet<int>();
```

Comment:
To consider: I find the comments horizontally between constructructors difficult to read for optical reasons. I'd move the comments 4 tabs further. There are just too many colors too close in the 2nd half:
![image](https://github.com/SonarSource/sonar-dotnet/assets/57188685/02a53c80-3b5f-425b-b853-d1705497cfb7)